[mHTTP GET / 200 [2.26, 127.0.0.1:31251][0m
[mHTTP GET /static/css/style.css 200 [2.64, 127.0.0.1:31259][0m
[mHTTP GET /static/css/login.css 200 [0.06, 127.0.0.1:31259][0m
[mHTTP GET /static/css/create.css 200 [0.03, 127.0.0.1:31259][0m
[mHTTP GET /static/css/brewing.css 200 [0.03, 127.0.0.1:31259][0m
[mHTTP GET /static/css/home.css 200 [0.02, 127.0.0.1:31259][0m
[mHTTP GET /static/css/template.css 200 [0.05, 127.0.0.1:31259][0m
[mHTTP GET /static/img/beer_background.webm 200 [0.67, 127.0.0.1:31259][0m
[mHTTP GET /static/img/favicon.ico 200 [0.06, 127.0.0.1:31268][0m
[32mHTTP GET /home 302 [0.19, 127.0.0.1:31276][0m
[mHTTP GET /login 200 [0.53, 127.0.0.1:31276][0m
[32mHTTP POST /login 302 [1.46, 127.0.0.1:31286][0m
[mHTTP GET /home 200 [0.25, 127.0.0.1:31286][0m
[mHTTP GET /static/js/home.js 200 [0.10, 127.0.0.1:31286][0m
[32mHTTP POST /brewing 302 [0.07, 127.0.0.1:31295][0m
[mHTTP GET /brewing 200 [0.15, 127.0.0.1:31295][0m
[mHTTP GET /static/img/download.png 200 [0.12, 127.0.0.1:31295][0m
[mHTTP GET /static/js/three.js 200 [0.50, 127.0.0.1:31296][0m
[mHTTP GET /static/js/server_connection.js 200 [0.39, 127.0.0.1:31295][0m
[mHTTP GET /static/js/protocol_csv.js 200 [0.37, 127.0.0.1:31301][0m
[mHTTP GET /static/js/createPDF.js 200 [0.58, 127.0.0.1:31302][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:31305]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:31305]
Fri Dec  2 17:42:12 2022: Another client connected
C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py changed, reloading.
[mHTTP GET /brewing 200 [0.27, 127.0.0.1:31462][0m
[32mHTTP POST /brewing 302 [0.07, 127.0.0.1:31470][0m
[mHTTP GET /brewing 200 [0.15, 127.0.0.1:31470][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:31473]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:31473]
Fri Dec  2 17:46:04 2022: Another client connected
C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py changed, reloading.
[mHTTP GET /brewing 200 [0.30, 127.0.0.1:31781][0m
[32mHTTP POST /brewing 302 [0.05, 127.0.0.1:31790][0m
[mHTTP GET /brewing 200 [0.10, 127.0.0.1:31790][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:31793]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:31793]
Fri Dec  2 17:49:19 2022: Another client connected
C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py changed, reloading.
[mHTTP GET /brewing 200 [0.40, 127.0.0.1:31955][0m
[32mHTTP POST /brewing 302 [0.06, 127.0.0.1:31965][0m
[mHTTP GET /brewing 200 [0.10, 127.0.0.1:31965][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:31970]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:31970]
Fri Dec  2 17:52:34 2022: Another client connected
[mHTTP GET /brewing 200 [0.64, 127.0.0.1:32905][0m
[32mHTTP POST /brewing 302 [0.04, 127.0.0.1:32918][0m
[mHTTP GET /brewing 200 [0.10, 127.0.0.1:32918][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:32922]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:32922]
Fri Dec  2 20:17:26 2022: Another client connected
C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\hardware.py changed, reloading.
[mHTTP GET /brewing 200 [0.24, 127.0.0.1:33041][0m
[32mHTTP POST /brewing 302 [0.05, 127.0.0.1:33050][0m
[mHTTP GET /brewing 200 [0.09, 127.0.0.1:33050][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:33054]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:33054]
Fri Dec  2 20:19:37 2022: Another client connected
[mHTTP GET /brewing 200 [1.20, 127.0.0.1:33240][0m
[32mHTTP POST /brewing 302 [0.06, 127.0.0.1:33240][0m
[mHTTP GET /brewing 200 [0.07, 127.0.0.1:33240][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:33250]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:33250]
Fri Dec  2 20:23:37 2022: Another client connected
C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py changed, reloading.
[mHTTP GET /brewing 200 [1.12, 127.0.0.1:33339][0m
[32mHTTP POST /brewing 302 [0.05, 127.0.0.1:33354][0m
[mHTTP GET /brewing 200 [0.09, 127.0.0.1:33354][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:33359]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:33359]
Fri Dec  2 20:25:36 2022: Another client connected
[mHTTP GET /brewing 200 [1.03, 127.0.0.1:33492][0m
[32mHTTP POST /brewing 302 [0.05, 127.0.0.1:33502][0m
[mHTTP GET /brewing 200 [0.09, 127.0.0.1:33502][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:33507]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:33507]
Fri Dec  2 20:28:30 2022: Another client connected
[mHTTP GET /brewing 200 [0.34, 127.0.0.1:33656][0m
[32mHTTP POST /brewing 302 [0.06, 127.0.0.1:33656][0m
[mHTTP GET /brewing 200 [0.11, 127.0.0.1:33656][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:33666]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:33666]
Fri Dec  2 20:31:19 2022: Another client connected
Exception inside application: send_json() takes 2 positional arguments but 3 were given
Traceback (most recent call last):
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\staticfiles.py", line 44, in __call__
    return await self.application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 71, in __call__
    return await application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 47, in __call__
    return await self.inner(dict(scope, cookies=cookies), receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 263, in __call__
    return await self.inner(wrapper.scope, receive, wrapper.send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\auth.py", line 185, in __call__
    return await super().__call__(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\middleware.py", line 26, in __call__
    return await self.inner(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 150, in __call__
    return await application(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 94, in app
    return await consumer(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 58, in __call__
    await await_many_dispatch(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\utils.py", line 51, in await_many_dispatch
    await dispatch(result)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 414, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\asyncio\tasks.py", line 442, in wait_for
    return await fut
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\concurrent\futures\thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\db.py", line 13, in thread_handler
    return super().thread_handler(loop, *args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 455, in thread_handler
    return func(*args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 125, in dispatch
    handler(message)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\generic\websocket.py", line 59, in websocket_receive
    self.receive(text_data=message["text"])
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 24, in receive
    self.evaluate_response(text_data_json)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 52, in evaluate_response
    if(brew_system.next_step(self)):
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 96, in next_step
    self.initiate_maischen(consumer)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 212, in initiate_maischen
    print(self.heat(int(json.loads(brew_recipe.objects.get(
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 250, in heat
    consumer.send_json(consumer,
TypeError: send_json() takes 2 positional arguments but 3 were given
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:33666]
[mHTTP GET /brewing 200 [0.12, 127.0.0.1:33737][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:33745]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:33745]
Fri Dec  2 20:32:57 2022: Another client connected
Exception inside application: send_json() takes 2 positional arguments but 3 were given
Traceback (most recent call last):
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\staticfiles.py", line 44, in __call__
    return await self.application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 71, in __call__
    return await application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 47, in __call__
    return await self.inner(dict(scope, cookies=cookies), receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 263, in __call__
    return await self.inner(wrapper.scope, receive, wrapper.send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\auth.py", line 185, in __call__
    return await super().__call__(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\middleware.py", line 26, in __call__
    return await self.inner(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 150, in __call__
    return await application(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 94, in app
    return await consumer(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 58, in __call__
    await await_many_dispatch(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\utils.py", line 51, in await_many_dispatch
    await dispatch(result)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 414, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\asyncio\tasks.py", line 442, in wait_for
    return await fut
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\concurrent\futures\thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\db.py", line 13, in thread_handler
    return super().thread_handler(loop, *args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 455, in thread_handler
    return func(*args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 125, in dispatch
    handler(message)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\generic\websocket.py", line 59, in websocket_receive
    self.receive(text_data=message["text"])
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 24, in receive
    self.evaluate_response(text_data_json)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 52, in evaluate_response
    if(brew_system.next_step(self)):
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 96, in next_step
    self.initiate_maischen(consumer)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 212, in initiate_maischen
    print(self.heat(int(json.loads(brew_recipe.objects.get(
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 250, in heat
    consumer.send_json(consumer,
TypeError: send_json() takes 2 positional arguments but 3 were given
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:33745]
[mHTTP GET /brewing 200 [0.09, 127.0.0.1:33763][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:33769]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:33769]
Fri Dec  2 20:33:27 2022: Another client connected
[mHTTP GET /brewing 200 [0.11, 127.0.0.1:33775][0m
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:33769]
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:33779]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:33779]
Fri Dec  2 20:33:37 2022: Another client connected
Exception inside application: send_json() takes 2 positional arguments but 3 were given
Traceback (most recent call last):
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\staticfiles.py", line 44, in __call__
    return await self.application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 71, in __call__
    return await application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 47, in __call__
    return await self.inner(dict(scope, cookies=cookies), receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 263, in __call__
    return await self.inner(wrapper.scope, receive, wrapper.send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\auth.py", line 185, in __call__
    return await super().__call__(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\middleware.py", line 26, in __call__
    return await self.inner(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 150, in __call__
    return await application(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 94, in app
    return await consumer(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 58, in __call__
    await await_many_dispatch(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\utils.py", line 51, in await_many_dispatch
    await dispatch(result)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 414, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\asyncio\tasks.py", line 442, in wait_for
    return await fut
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\concurrent\futures\thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\db.py", line 13, in thread_handler
    return super().thread_handler(loop, *args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 455, in thread_handler
    return func(*args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 125, in dispatch
    handler(message)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\generic\websocket.py", line 59, in websocket_receive
    self.receive(text_data=message["text"])
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 24, in receive
    self.evaluate_response(text_data_json)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 52, in evaluate_response
    if(brew_system.next_step(self)):
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 96, in next_step
    self.initiate_maischen(consumer)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 212, in initiate_maischen
    print(self.heat(int(json.loads(brew_recipe.objects.get(
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 250, in heat
    consumer.send_json(consumer,
TypeError: send_json() takes 2 positional arguments but 3 were given
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:33779]
[mHTTP GET /brewing 200 [1.31, 127.0.0.1:33876][0m
[32mHTTP POST /brewing 302 [0.06, 127.0.0.1:33876][0m
[mHTTP GET /brewing 200 [0.15, 127.0.0.1:33876][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:33888]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:33888]
Fri Dec  2 20:35:35 2022: Another client connected
Exception inside application: send_json() takes 2 positional arguments but 3 were given
Traceback (most recent call last):
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\staticfiles.py", line 44, in __call__
    return await self.application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 71, in __call__
    return await application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 47, in __call__
    return await self.inner(dict(scope, cookies=cookies), receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 263, in __call__
    return await self.inner(wrapper.scope, receive, wrapper.send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\auth.py", line 185, in __call__
    return await super().__call__(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\middleware.py", line 26, in __call__
    return await self.inner(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 150, in __call__
    return await application(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 94, in app
    return await consumer(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 58, in __call__
    await await_many_dispatch(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\utils.py", line 51, in await_many_dispatch
    await dispatch(result)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 414, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\asyncio\tasks.py", line 442, in wait_for
    return await fut
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\concurrent\futures\thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\db.py", line 13, in thread_handler
    return super().thread_handler(loop, *args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 455, in thread_handler
    return func(*args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 125, in dispatch
    handler(message)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\generic\websocket.py", line 59, in websocket_receive
    self.receive(text_data=message["text"])
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 24, in receive
    self.evaluate_response(text_data_json)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 52, in evaluate_response
    if(brew_system.next_step(self)):
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 96, in next_step
    self.initiate_maischen(consumer)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 212, in initiate_maischen
    print(self.heat(int(json.loads(brew_recipe.objects.get(
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 250, in heat
    consumer.send_json(consumer,
TypeError: send_json() takes 2 positional arguments but 3 were given
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:33888]
[mHTTP GET /brewing 200 [0.19, 127.0.0.1:33919][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:33926]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:33926]
Fri Dec  2 20:36:13 2022: Another client connected
Exception inside application: send_json() takes 2 positional arguments but 3 were given
Traceback (most recent call last):
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\staticfiles.py", line 44, in __call__
    return await self.application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 71, in __call__
    return await application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 47, in __call__
    return await self.inner(dict(scope, cookies=cookies), receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 263, in __call__
    return await self.inner(wrapper.scope, receive, wrapper.send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\auth.py", line 185, in __call__
    return await super().__call__(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\middleware.py", line 26, in __call__
    return await self.inner(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 150, in __call__
    return await application(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 94, in app
    return await consumer(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 58, in __call__
    await await_many_dispatch(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\utils.py", line 51, in await_many_dispatch
    await dispatch(result)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 414, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\asyncio\tasks.py", line 442, in wait_for
    return await fut
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\concurrent\futures\thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\db.py", line 13, in thread_handler
    return super().thread_handler(loop, *args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 455, in thread_handler
    return func(*args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 125, in dispatch
    handler(message)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\generic\websocket.py", line 59, in websocket_receive
    self.receive(text_data=message["text"])
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 24, in receive
    self.evaluate_response(text_data_json)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 52, in evaluate_response
    if(brew_system.next_step(self)):
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 96, in next_step
    self.initiate_maischen(consumer)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 212, in initiate_maischen
    print(self.heat(int(json.loads(brew_recipe.objects.get(
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 250, in heat
    consumer.send_json(consumer,
TypeError: send_json() takes 2 positional arguments but 3 were given
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:33926]
C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py changed, reloading.
[mHTTP GET /brewing 200 [0.73, 127.0.0.1:34056][0m
[32mHTTP POST /brewing 302 [0.04, 127.0.0.1:34064][0m
[mHTTP GET /brewing 200 [0.09, 127.0.0.1:34064][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:34069]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:34069]
Fri Dec  2 20:39:01 2022: Another client connected
Exception inside application: external_send() takes 2 positional arguments but 3 were given
Traceback (most recent call last):
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\staticfiles.py", line 44, in __call__
    return await self.application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 71, in __call__
    return await application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 47, in __call__
    return await self.inner(dict(scope, cookies=cookies), receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 263, in __call__
    return await self.inner(wrapper.scope, receive, wrapper.send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\auth.py", line 185, in __call__
    return await super().__call__(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\middleware.py", line 26, in __call__
    return await self.inner(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 150, in __call__
    return await application(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 94, in app
    return await consumer(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 58, in __call__
    await await_many_dispatch(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\utils.py", line 51, in await_many_dispatch
    await dispatch(result)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 414, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\asyncio\tasks.py", line 442, in wait_for
    return await fut
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\concurrent\futures\thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\db.py", line 13, in thread_handler
    return super().thread_handler(loop, *args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 455, in thread_handler
    return func(*args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 125, in dispatch
    handler(message)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\generic\websocket.py", line 59, in websocket_receive
    self.receive(text_data=message["text"])
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 24, in receive
    self.evaluate_response(text_data_json)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 52, in evaluate_response
    if(brew_system.next_step(self)):
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 96, in next_step
    self.initiate_maischen(consumer)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 212, in initiate_maischen
    print(self.heat(int(json.loads(brew_recipe.objects.get(
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py", line 250, in heat
    consumer.external_send(consumer,
TypeError: external_send() takes 2 positional arguments but 3 were given
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:34069]
C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py changed, reloading.
C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py changed, reloading.
C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py changed, reloading.
[mHTTP GET /brewing 200 [0.68, 127.0.0.1:34173][0m
[32mHTTP POST /brewing 302 [0.10, 127.0.0.1:34173][0m
[mHTTP GET /brewing 200 [0.16, 127.0.0.1:34173][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:34191]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:34191]
Fri Dec  2 20:40:43 2022: Another client connected
C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py changed, reloading.
[mHTTP GET /brewing 200 [1.05, 127.0.0.1:34314][0m
[32mHTTP POST /brewing 302 [0.06, 127.0.0.1:34322][0m
[mHTTP GET /brewing 200 [0.09, 127.0.0.1:34322][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:34327]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:34327]
Fri Dec  2 20:43:03 2022: Another client connected
[mHTTP GET /brewing 200 [0.92, 127.0.0.1:34525][0m
[32mHTTP POST /brewing 302 [0.03, 127.0.0.1:34536][0m
[mHTTP GET /brewing 200 [0.07, 127.0.0.1:34536][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:34543]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:34543]
Fri Dec  2 20:46:56 2022: Another client connected
[mHTTP GET /brewing 200 [0.90, 127.0.0.1:34688][0m
[32mHTTP POST /brewing 302 [0.05, 127.0.0.1:34696][0m
[mHTTP GET /brewing 200 [0.10, 127.0.0.1:34696][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:34702]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:34702]
Fri Dec  2 20:49:48 2022: Another client connected
[mHTTP GET /brewing 200 [0.15, 127.0.0.1:34883][0m
[mHTTP GET /brewing 200 [0.06, 127.0.0.1:34901][0m
[32mHTTP POST /brewing 302 [0.05, 127.0.0.1:34901][0m
[mHTTP GET /brewing 200 [0.13, 127.0.0.1:34901][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:34907]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:34907]
Fri Dec  2 20:54:37 2022: Another client connected
[mHTTP GET /brewing 200 [0.24, 127.0.0.1:34992][0m
[32mHTTP POST /brewing 302 [0.05, 127.0.0.1:35000][0m
[mHTTP GET /brewing 200 [0.08, 127.0.0.1:35000][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:35005]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:35005]
Fri Dec  2 20:56:22 2022: Another client connected
C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py changed, reloading.
[mHTTP GET /brewing 200 [0.19, 127.0.0.1:35093][0m
[32mHTTP POST /brewing 302 [0.04, 127.0.0.1:35100][0m
[mHTTP GET /brewing 200 [0.09, 127.0.0.1:35100][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:35105]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:35105]
Fri Dec  2 20:58:21 2022: Another client connected
C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py changed, reloading.
[mHTTP GET /brewing 200 [0.22, 127.0.0.1:35186][0m
[32mHTTP POST /brewing 302 [0.05, 127.0.0.1:35186][0m
[mHTTP GET /brewing 200 [0.08, 127.0.0.1:35186][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:35194]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:35194]
Fri Dec  2 21:00:03 2022: Another client connected
C:\Users\Jonas\Documents\django\brew_seminar\brewing\process\brew_server.py changed, reloading.
[mHTTP GET /brewing 200 [1.23, 127.0.0.1:35355][0m
[32mHTTP POST /brewing 302 [0.04, 127.0.0.1:35366][0m
[mHTTP GET /brewing 200 [0.09, 127.0.0.1:35366][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:35372]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:35372]
Fri Dec  2 21:03:33 2022: Another client connected
[mHTTP GET /brewing 200 [0.12, 127.0.0.1:35381][0m
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:35372]
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:35385]
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:35385]
[mHTTP GET /brewing 200 [0.19, 127.0.0.1:35474][0m
[32mHTTP POST /brewing 302 [0.05, 127.0.0.1:35482][0m
[mHTTP GET /brewing 200 [0.09, 127.0.0.1:35482][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:35486]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:35486]
Fri Dec  2 21:05:36 2022: Another client connected
[mHTTP GET /brewing 200 [0.46, 127.0.0.1:35525][0m
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:35486]
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:35535]
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:35535]
Application instance <Task pending name='Task-4' coro=<StaticFilesWrapper.__call__() running at C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\staticfiles.py:44> wait_for=<Future pending cb=[_chain_future.<locals>._call_check_cancel() at C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\asyncio\futures.py:384, <TaskWakeupMethWrapper object at 0x00000191722755E0>()]>> for connection <WebSocketProtocol client=['127.0.0.1', 35486] path=b'/ws/socket-server/'> took too long to shut down and was killed.
[mHTTP GET /brewing 200 [0.10, 127.0.0.1:35544][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:35548]
Application instance <Task pending name='Task-41' coro=<StaticFilesWrapper.__call__() running at C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\staticfiles.py:44> wait_for=<Future pending cb=[_chain_future.<locals>._call_check_cancel() at C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\asyncio\futures.py:384, <TaskWakeupMethWrapper object at 0x00000191722F7BE0>()]>> for connection <WebSocketProtocol client=['127.0.0.1', 35535] path=b'/ws/socket-server/'> took too long to shut down and was killed.
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:35548]
[mHTTP GET /brewing 200 [0.49, 127.0.0.1:35594][0m
[32mHTTP POST /brewing 302 [0.05, 127.0.0.1:35601][0m
[mHTTP GET /brewing 200 [0.08, 127.0.0.1:35601][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:35606]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:35606]
Fri Dec  2 21:07:25 2022: Another client connected
[mHTTP GET /brewing 200 [0.18, 127.0.0.1:35769][0m
[32mHTTP POST /brewing 302 [0.17, 127.0.0.1:35769][0m
[mHTTP GET /brewing 200 [0.12, 127.0.0.1:35769][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:35779]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:35779]
Fri Dec  2 21:10:25 2022: Another client connected
[mHTTP GET /brewing 200 [0.15, 127.0.0.1:35807][0m
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:35779]
Application instance <Task pending name='Task-4' coro=<StaticFilesWrapper.__call__() running at C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\staticfiles.py:44> wait_for=<Future pending cb=[_chain_future.<locals>._call_check_cancel() at C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\asyncio\futures.py:384, <TaskWakeupMethWrapper object at 0x0000018FBCADFA60>()]>> for connection <WebSocketProtocol client=['127.0.0.1', 35779] path=b'/ws/socket-server/'> took too long to shut down and was killed.
[mHTTP GET /brewing 200 [0.37, 127.0.0.1:35891][0m
[mHTTP GET /static/css/style.css 200 [1.78, 127.0.0.1:35891][0m
[mHTTP GET /static/img/favicon.ico 200 [0.02, 127.0.0.1:35891][0m
[32mHTTP POST /brewing 302 [0.04, 127.0.0.1:35900][0m
[mHTTP GET /brewing 200 [0.10, 127.0.0.1:35900][0m
[mHTTP GET /static/js/three.js 200 [0.18, 127.0.0.1:35900][0m
[mHTTP GET /static/js/server_connection.js 200 [0.02, 127.0.0.1:35900][0m
[mHTTP GET /static/js/protocol_csv.js 200 [0.04, 127.0.0.1:35900][0m
[mHTTP GET /static/js/createPDF.js 200 [0.18, 127.0.0.1:35901][0m
[mHTTP GET /static/img/download.png 200 [0.27, 127.0.0.1:35900][0m
[mHTTP GET /brewing 200 [0.10, 127.0.0.1:35914][0m
[mHTTP GET /brewing 200 [0.38, 127.0.0.1:36119][0m
[mHTTP GET /static/css/style.css 200 [0.06, 127.0.0.1:36119][0m
[mHTTP GET /static/css/login.css 200 [0.13, 127.0.0.1:36146][0m
[mHTTP GET /static/js/three.js 200 [0.37, 127.0.0.1:36147][0m
[mHTTP GET /static/css/create.css 200 [0.31, 127.0.0.1:36146][0m
[mHTTP GET /static/js/server_connection.js 200 [0.05, 127.0.0.1:36146][0m
[mHTTP GET /static/css/brewing.css 200 [0.11, 127.0.0.1:36150][0m
[mHTTP GET /static/js/protocol_csv.js 200 [0.06, 127.0.0.1:36150][0m
[mHTTP GET /static/css/home.css 200 [0.15, 127.0.0.1:36146][0m
[mHTTP GET /static/js/createPDF.js 200 [0.11, 127.0.0.1:36150][0m
[mHTTP GET /static/css/template.css 200 [0.09, 127.0.0.1:36146][0m
[mHTTP GET /static/img/download.png 200 [0.04, 127.0.0.1:36146][0m
[mHTTP GET /static/img/favicon.ico 200 [0.02, 127.0.0.1:36146][0m
[mHTTP GET /brewing 200 [0.21, 127.0.0.1:36206][0m
[mHTTP GET /brewing 200 [0.20, 127.0.0.1:36299][0m
[mHTTP GET /static/js/server_connection.js 200 [0.14, 127.0.0.1:36299][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:36308]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:36308]
Fri Dec  2 21:16:02 2022: Another client connected
[mHTTP GET /brewing 200 [0.31, 127.0.0.1:36361][0m
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:36308]
[mHTTP GET /static/js/server_connection.js 200 [0.03, 127.0.0.1:36370][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:36376]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:36376]
Fri Dec  2 21:17:21 2022: Another client connected
[mHTTP GET /brewing 200 [0.16, 127.0.0.1:36398][0m
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:36376]
[mHTTP GET /static/js/server_connection.js 200 [0.03, 127.0.0.1:36412][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:36415]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:36415]
Fri Dec  2 21:17:55 2022: Another client connected
[mHTTP GET /brewing 200 [0.45, 127.0.0.1:36497][0m
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:36415]
[mHTTP GET /static/js/server_connection.js 200 [0.02, 127.0.0.1:36497][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:36504]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:36504]
Fri Dec  2 21:19:20 2022: Another client connected
[mHTTP GET /brewing 200 [1.05, 127.0.0.1:36589][0m
[mHTTP GET /static/js/home.js 200 [2.79, 127.0.0.1:36589][0m
[32mHTTP POST /brewing 302 [0.05, 127.0.0.1:36597][0m
[mHTTP GET /brewing 200 [0.12, 127.0.0.1:36597][0m
[36mHTTP GET /static/js/server_connection.js 304 [0.02, 127.0.0.1:36597][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:36603]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:36603]
Fri Dec  2 21:21:18 2022: Another client connected
[mHTTP GET /brewing 200 [0.97, 127.0.0.1:36659][0m
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:36603]
Exception inside application: heat() missing 1 required positional argument: 'consumer'
Traceback (most recent call last):
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\staticfiles.py", line 44, in __call__
    return await self.application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 71, in __call__
    return await application(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 47, in __call__
    return await self.inner(dict(scope, cookies=cookies), receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\sessions.py", line 263, in __call__
    return await self.inner(wrapper.scope, receive, wrapper.send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\auth.py", line 185, in __call__
    return await super().__call__(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\middleware.py", line 26, in __call__
    return await self.inner(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\routing.py", line 150, in __call__
    return await application(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 94, in app
    return await consumer(scope, receive, send)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 58, in __call__
    await await_many_dispatch(
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\utils.py", line 51, in await_many_dispatch
    await dispatch(result)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 414, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\asyncio\tasks.py", line 442, in wait_for
    return await fut
  File "C:\Users\Jonas\AppData\Local\Programs\Python\Python39\lib\concurrent\futures\thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\db.py", line 13, in thread_handler
    return super().thread_handler(loop, *args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\asgiref\sync.py", line 455, in thread_handler
    return func(*args, **kwargs)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\consumer.py", line 125, in dispatch
    handler(message)
  File "C:\Users\Jonas\Documents\django\django_virtual\lib\site-packages\channels\generic\websocket.py", line 59, in websocket_receive
    self.receive(text_data=message["text"])
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 24, in receive
    self.evaluate_response(text_data_json)
  File "C:\Users\Jonas\Documents\django\brew_seminar\brewing\consumer.py", line 72, in evaluate_response
    brew_system.heat(0)
TypeError: heat() missing 1 required positional argument: 'consumer'
[36mHTTP GET /static/js/server_connection.js 304 [0.01, 127.0.0.1:36659][0m
WebSocket HANDSHAKING /ws/socket-server/ [127.0.0.1:36666]
WebSocket CONNECT /ws/socket-server/ [127.0.0.1:36666]
Fri Dec  2 21:22:51 2022: Another client connected
WebSocket DISCONNECT /ws/socket-server/ [127.0.0.1:36666]
